src/main/java/org/elasticsearch/index/engine/internal/InternalEngine.java

```diff
@@ -941,7 +927,7 @@ protected void doRun() throws Exception {
     public SnapshotIndexCommit snapshotIndex() throws EngineException {
         // we have to flush outside of the readlock otherwise we might have a problem upgrading
         // the to a write lock when we fail the engine in this operation
-        flush(FlushType.COMMIT, false, true);
+        flush(false, false, true);
```

Comment:
yes I think so - I was wondering too why we can't flush translog here we can at least try?